library/item/item_group/src/main/java/org/quiltmc/qsl/item/group/mixin/client/CreativeInventoryScreenMixin.java

```diff
@@ -165,6 +174,19 @@ private void renderTabIcon(MatrixStack matrixStack, ItemGroup itemGroup, Callbac
 		if (this.quilt$isGroupNotVisible(itemGroup)) {
 			info.cancel();
 		}
+
+		this.quilt$renderTabIcon$matrices = matrixStack;
+		this.quilt$renderTabIcon$itemGroup = itemGroup;
+	}
+
+	@Redirect(method = "renderTabIcon", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/render/item/ItemRenderer;renderInGuiWithOverrides(Lnet/minecraft/item/ItemStack;II)V"))
```

Comment:
Can we use a ModifyArgs here instead? (And just change the item arg to render an empty itemstack). This just seems like something that others might be liable to target (if they were already doing custom rendering, or on fabric mods)

Reply:
Can't use `ModifyArgs` here because there is no guarantee, that the item group that has a custom icon renderer doesn't have a regular icon supplier and I don't want the custom icon renderer to render over or under the regular item icon, so I have to use `Redirect` to cancel that vanilla behaviour if the item group has a custom icon renderer

Reply:
I could use a `Inject` with a `CallbackInfo.cancel` if you are really worried about mod compatibility.

Reply:
Oh crap lmao I didn't read the `And just change the item arg to render an empty itemstack`, oopsie doopsie will change it now!

Reply:
Resolved.